rust/onnxruntime-sys/build.rs

```diff
@@ -0,0 +1,420 @@
+    let mut archive = tar::Archive::new(tar);
+    archive.unpack(output).unwrap();
+}
+
+fn extract_zip(filename: &Path, outpath: &Path) {
+    let file = fs::File::open(&filename).unwrap();
+    let buf = io::BufReader::new(file);
+    let mut archive = zip::ZipArchive::new(buf).unwrap();
+    for i in 0..archive.len() {
+        let mut file = archive.by_index(i).unwrap();
+        #[allow(deprecated)]
+        let outpath = outpath.join(file.sanitized_name());
+        if !file.name().ends_with('/') {
+            println!(
+                "File {} extracted to \"{}\" ({} bytes)",
+                i,
+                outpath.as_path().display(),
+                file.size()
+            );
+            if let Some(p) = outpath.parent() {
+                if !p.exists() {
+                    fs::create_dir_all(&p).unwrap();
+                }
+            }
+            let mut outfile = fs::File::create(&outpath).unwrap();
+            io::copy(&mut file, &mut outfile).unwrap();
+        }
+    }
+}
+
+trait OnnxPrebuiltArchive {
+    fn as_onnx_str(&self) -> Cow<str>;
+}
+
+#[derive(Debug)]
+enum Architecture {
+    X86,
+    X86_64,
+    Arm,
+    Arm64,
+}
+
+impl FromStr for Architecture {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "x86" => Ok(Architecture::X86),
+            "x86_64" => Ok(Architecture::X86_64),
+            "arm" => Ok(Architecture::Arm),
+            "aarch64" => Ok(Architecture::Arm64),
+            _ => Err(format!("Unsupported architecture: {}", s)),
+        }
+    }
+}
+
+impl OnnxPrebuiltArchive for Architecture {
+    fn as_onnx_str(&self) -> Cow<str> {
+        match self {
+            Architecture::X86 => Cow::from("x86"),
+            Architecture::X86_64 => Cow::from("x64"),
+            Architecture::Arm => Cow::from("arm"),
+            Architecture::Arm64 => Cow::from("arm64"),
+        }
+    }
+}
+
+#[derive(Debug)]
+#[allow(clippy::enum_variant_names)]
+enum Os {
+    Windows,
+    Linux,
+    MacOs,
+}
+
+impl Os {
+    fn archive_extension(&self) -> &'static str {
+        match self {
+            Os::Windows => "zip",
+            Os::Linux => "tgz",
+            Os::MacOs => "tgz",
+        }
+    }
+}
+
+impl FromStr for Os {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "windows" => Ok(Os::Windows),
+            "macos" => Ok(Os::MacOs),
+            "linux" => Ok(Os::Linux),
+            _ => Err(format!("Unsupported os: {}", s)),
+        }
+    }
+}
+
+impl OnnxPrebuiltArchive for Os {
+    fn as_onnx_str(&self) -> Cow<str> {
+        match self {
+            Os::Windows => Cow::from("win"),
+            Os::Linux => Cow::from("linux"),
+            Os::MacOs => Cow::from("osx"),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+enum Accelerator {
+    None,
+    Gpu,
+}
```

Comment:
```suggestion
enum Accelerator {
    Cpu,
    Cuda,
}
```

It seems [onnxruntime provide multiple type accelerator](https://onnxruntime.ai/) .
Currently provided binaries are directml and cuda only. 
However, considering onnxruntime may provide any accelerator in the future, Accelerator name change.

Reply:
Yeah, sounds good. Cuda is only one way to use the GPU.